cura/CuraApplication.py

```diff
@@ -723,6 +738,20 @@ def preRun(self):
     def run(self):
         self.preRun()
 
+        container_registry = ContainerRegistry.getInstance()
+        self._variant_manager = VariantManager(container_registry)
+        self._variant_manager.initialize()
+
+        from cura.Machines.MaterialManager import MaterialManager
+        self._material_manager = MaterialManager(container_registry, parent = self)
+        self._material_manager.initialize()
+
+        from cura.Machines.QualityManager import QualityManager
+        self._quality_manager = QualityManager(container_registry, parent = self)
+        self._quality_manager.initialize()
+
+        self.getMachineManager()  # ensure creation of machine manager
```

Comment:
getMachineManager() actually creates a MachineManager if it's not there yet. It used to be created on demand, which means it will be created whenever is needed, but that actually breaks things because of the uncertainty when it will be created and whether it will be (properly) initialized before being used. This function call is not clear what it does. I will change it to something more explicit.

Reply:
Changed in 0b8678d2a